test/relayclient/RevertMessage.test.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  TestRecipientInstance
+} from '../../types/truffle-contracts'
+import { Address } from '../../src/relayclient/types/Aliases'
+import { GsnTestEnvironment } from '../../src/relayclient/GsnTestEnvironment'
+
+const underlyingProvider = web3.currentProvider as HttpProvider
+
+describe('RevertMessage.test', function () {
+  let web3: Web3
+  let forwarderAddress: Address
+  let relayHubAddress: Address
+  let paymasterAddress: Address
+  let relayProvider: RelayProvider
+
+  before(async function () {
+    this.timeout(30000)
+    web3 = new Web3(underlyingProvider)
+
+    // Addresses copied from latest release: https://github.com/opengsn/gsn/releases/tag/v2.0.1
+    const networks: any = {
+      3: {
+        hub: '0x29e41C2b329fF4921d8AC654CEc909a0B575df20',
+        forwarder: '0x25CEd1955423BA34332Ec1B60154967750a0297D',
+        paymaster: '0x8057c0fb7089BB646f824fF4A4f5a18A8d978ecC'
+      },
+      4: {
+        hub: '0x53C88539C65E0350408a2294C4A85eB3d8ce8789',
+        forwarder: '0x956868751Cc565507B3B58E53a6f9f41B56bed74',
+        paymaster: '0x43d66E6Dce20264F6511A0e8EEa3f570980341a2'
+      },
+      42: {
+        hub: '0xE9dcD2CccEcD77a92BA48933cb626e04214Edb92',
+        forwarder: '0x0842Ad6B8cb64364761C7c170D0002CC56b1c498',
+        paymaster: '0x083082b7Eada37dbD8f263050570B31448E61c94'
+      }
+    }
+    const chain = await web3.eth.net.getId()
+    if (chain > 1000) {
+      // @ts-ignore
+      const { deploymentResult } = await GsnTestEnvironment.startGsn(web3.currentProvider?.host ?? 'localhost')
+      forwarderAddress = deploymentResult.forwarderAddress
+      relayHubAddress = deploymentResult.relayHubAddress
+      paymasterAddress = deploymentResult.naivePaymasterAddress
+    } else {
+      relayHubAddress = networks[chain].hub
+      paymasterAddress = networks[chain].paymaster
+      forwarderAddress = networks[chain].forwarder
+    }
+  })
+
+  after('after all', async function () {
+    await GsnTestEnvironment.stopGsn()
+  })
+
+  describe('Use Provider to relay request', () => {
+    let testRecipient: TestRecipientInstance
+    let gasLess: Address
+
+    before(async function () {
+      this.timeout(30000)
+      const TestRecipient = artifacts.require('TestRecipient')
+      testRecipient = await TestRecipient.new(forwarderAddress)
+      const gsnConfig = configureGSN({
+        logLevel: 5,
+        relayHubAddress,
+        paymasterAddress
+      })
+
+      relayProvider = new RelayProvider(underlyingProvider as any, gsnConfig)
+      // NOTE: in real application its enough to set the provider in web3.
+      // however, in Truffle, all contracts are built BEFORE the test have started, and COPIED the web3,
+      // so changing the global one is not enough.
+      // @ts-ignore
+      TestRecipient.web3.setProvider(relayProvider)
+
+      gasLess = relayProvider.newAccount().address
+    })
+
+    it('should show relayCall revert messages', async function () {
+      this.timeout(30000)
```

Comment:
Why does this simple test require a 30 seconds timeout?

Reply:
it isn't. but on ropsten it sometimes take more than default 2 seconds. I didn't calculate how much timeout is good enough...